src/DateRange/DateRange.tsx

```diff
@@ -0,0 +1,264 @@
+
+	React.useEffect(() => {
+		setDateRange({ dateFrom, dateTo });
+	}, []);
+
+	const handleSelectedNoFilter = () => {
+		setPrevValue(value);
+		setValue(DateRangeValueTypes.NoFilter);
+		onChange({ type: DateRangeValueTypes.NoFilter });
+	};
+
+	const handleSelectedLast = (currentValue: LastPeriods) => () => {
+		setPrevValue(value);
+		setValue(currentValue);
+		onChange({ type: DateRangeValueTypes.Last, period: currentValue });
+	};
+
+	const onCloseFilter = () => {
+		setShowFilter(false);
+		setValue(prevValue);
+
+		setDateFrom(dateRange.dateFrom);
+		setDateTo(dateRange.dateTo);
+	};
+
+	const handleToggleFilter = (currentValue: DateRangeValueTypes) => () => {
+		setPrevValue(value);
+		setValue(currentValue);
+		setShowFilter(prev => !prev);
+	};
+
+	const handleApplyFilter = () => {
+		setShowFilter(false);
+		setDateRange({ dateFrom, dateTo });
+		onChange({ type: DateRangeValueTypes.Concrete, dateFrom, dateTo });
+	};
+
+	const handleChangeDateFrom = (currentDateFrom: Date) => {
+		setDateFrom(currentDateFrom);
+	};
+
+	const handleChangeDateTo = (currentDateTo: Date) => {
+		setDateTo(currentDateTo);
+	};
+
+	const handleChangeTimeFrom = (hours: number, minutes: number) => {
+		setTimeFrom({ hours, minutes });
+	};
+
+	const handleChangeTimeTo = (hours: number, minutes: number) => {
+		setTimeTo({ hours, minutes });
+	};
+
+	const InnerEditorComponent = () => (
+		<div className="kit-date-range__popup">
+			<div className="kit-date-range__popup-dates">
+				<span className="kit-date-range__popup-date-text">
+					{radioConcreteFromText}
+				</span>
+				<div className="kit-date-range__popup-fields">
+					<div className="kit-date-range__popup-date">
+						<DateField
+							error={isError}
+							defaultDate={dateFrom}
+							onChange={handleChangeDateFrom}
+						/>
+					</div>
+					<div className="kit-date-range__popup-date">
+						<TimeField
+							error={isError}
+							hours={timeFrom.hours}
+							minutes={timeFrom.minutes}
+							onChange={handleChangeTimeFrom}
+						/>
+					</div>
+				</div>
+			</div>
+			<div className="kit-date-range__popup-dates">
+				<span className="kit-date-range__popup-date-text">
+					{radioConcreteToText}
+				</span>
+				<div className="kit-date-range__popup-fields">
+					<div className="kit-date-range__popup-date">
+						<DateField
+							error={isError}
+							defaultDate={dateTo}
+							onChange={handleChangeDateTo}
+						/>
+					</div>
+					<div className="kit-date-range__popup-date">
+						<TimeField
+							error={isError}
+							hours={timeTo.hours}
+							minutes={timeTo.minutes}
+							onChange={handleChangeTimeTo}
+						/>
+					</div>
+				</div>
+			</div>
+			{isError && (
+				<Tooltip
+					title={<IconSvg type="warning" />}
+					position="top"
+					className="kit-date-range__popup-tooltip"
+				>
+					{tooltipContent}
+				</Tooltip>
+			)}
+		</div>
+	);
+
+	const isError = dateFrom > dateTo;
```

Comment:
Валидация не работает со временем в одной дате.
```suggestion
	const hasError = dateFrom > dateTo;
```
И лучше наверх вынести